### ISSUE TITLE
Custom ML Model Drastically Different Results

### ISSUE THREAD

**Issue by kossyrev-bg** [AxisCommunications/acap-native-sdk-examples#359]
> **Please do not disclose security vulnerabilities as issues. See our [security policy](../../SECURITY.md) for responsible disclosures.**

### Describe the bug

We trained a custom 2-class model based on mobilenet and uploaded to the camera using roughly the same setup as the object detection example. Once uploaded, the model gave completely the opposite detection from what was expected. Model input is [1,224,224,3] uint8. Model output is [1, 2] float32.

What's even weirder is we took the exact image from the camera that was used for the classification, ran it through the same model in python, and got results that made sense!

What's even weirder x2 is that when I use `larod-client` to run the model on the camera on a jpg, it outputs logits that make a lot more sense. So I'm suspecting that it has something to do with either our C code or larod.

### To reproduce

1. Upload an eap built using the object detection example using our model (can provide link to unweighted model)
2. Note the model outputs using app log
3. Open a Jupyter notebook on dev machine, load in the model using the `tf.lite.Interpreter` module
4. Run the model on the same image and note the vastly different logit outputs

### Environment

- Axis device model: M3067-P
- Axis device firmware version: 11.11.135
- Stack trace or logs: 
AXIS camera:
```
2025-03-17T11:36:30.928-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Converted image in 2 ms
2025-03-17T11:36:31.329-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Ran inference for 402 ms
2025-03-17T11:36:31.336-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Image successfully saved as JPEG: /tmp/actual.jpg
2025-03-17T11:36:31.336-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Last inference sku presence: 1
2025-03-17T11:36:31.336-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Last inference time: 2025-03-17 15:36:30.928
2025-03-17T11:36:31.337-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: Logits: -1.375955, 1.545594
2025-03-17T11:36:31.337-04:00 axis-b8a44f4ba9eb [ INFO    ] object_detection[5099]: SKU presence: true
```
Python logs from model output:
```
array([[ 1.7717782, -1.2063171]], dtype=float32)
```
- OS and version: Ubuntu 24.04
- Version: tensorflow==2.10.1

**Comment by kossyrev-bg** [comment]
Closing, as it seems we were incorrectly pre-processing data before feeding it to our model
